src/libraries/System.Runtime.Numerics/src/System/Numerics/BigInteger.cs

```diff
@@ -2303,6 +2303,37 @@ private static BigInteger Subtract(uint[]? leftBits, int leftSign, uint[]? right
             return !right.Equals(left);
         }
 
+        /// <summary>
+        /// Gets the number of bits required to represent this BigInteger instance in two's complement notation without the sign bit.
+        /// </summary>
+        /// <returns>The non-negative number of bits in two's complement notation without the sign bit.</returns>
+        /// <remarks>If <see cref="IsZero"/> is true, this method returns 0. For positive integers the return value is equal to the binary representation string length.</remarks>
+        public int GetBitLength()
+        {
+            AssertValid();
+
+            uint highValue;
+            int byteLength;
+            uint[]? bits = _bits;
+
+            if (bits == null)
+            {
+                byteLength = 1;
+
+                int sign = _sign;
+                highValue = sign < 0 ? (uint)-sign : (uint)sign;
+            }
+            else
+            {
+                byteLength = bits.Length;
+                highValue = bits[byteLength - 1];
+            }
+
+            int highZeros = NumericsHelpers.CbitHighZero(highValue);
```

Comment:
It would be good to use `BitOperations.LeadingZeroCount` instead.
Ideally we would switch all usages of `CbitHighZero` to `BitOperations.LeadingZeroCount` but that should likely be a separate PR.

Reply:
This is an optional recommendation, right?
Suggested options:
* edit it (edits by maintainers allowed)
* wait for me to apply the change somewhere in mid-July (unfortunately I've got no free time at the moment and I'm not familiar with `BitOperations` to do it right now without proper testing)
* merge as is, switching it to use `BitOperations.LeadingZeroCount` with the rest of the `CbitHighZero` usages

Reply:
Right, it isn't required. If you aren't able to handle this particular scenario I'll log a bug and have it done as part of fixing all `CbitHighZero` usages.

Reply:
Nvm, I've already rearranged my schedule. I'll try to do it right now.

Reply:
Turns out it's a direct replacement.